cdk/src/open-data-platform/data-plane/data-import/write-violations-data.handler.ts

```diff
@@ -0,0 +1,260 @@
+import { APIGatewayProxyEvent, APIGatewayProxyResult } from 'aws-lambda';
+import { ConnectionPool, sql } from '@databases/pg';
+import * as AWS from 'aws-sdk';
+import { connectToDb } from '../schema/schema.handler';
+
+const { chain } = require('stream-chain');
+const { streamArray } = require('stream-json/streamers/StreamArray');
+const Batch = require('stream-json/utils/Batch');
+const Pick = require('stream-json/filters/Pick');
+const moment = require('moment');
+
+const S3 = new AWS.S3();
+
+const DEFAULT_NUMBER_ROWS_TO_INSERT = 10;
+
+const VIOLATION = 'SDWISDM.VIOLATION';
+const VIOLATION_ID = `${VIOLATION}.VIOLATION_ID`;
+const PWSID = `${VIOLATION}.VIOLATION_ID`;
+const VIOLATION_CODE = `${VIOLATION}.VIOLATION_CODE`;
+const COMPLIANCE_STATUS_CODE = `${VIOLATION}.COMPLIANCE_STATUS_CODE`;
+const COMPL_PER_BEGIN_DATE = `${VIOLATION}.COMPL_PER_BEGIN_DATE`;
+
+/**
+ * Status codes that indicate a Lead and Copper Rule violation.
+ * See https://www.epa.gov/sites/default/files/2019-07/documents/environments-and-contaminants-methods-drinking-water.pdf
+ * Page 4.
+ */
+const LEAD_AND_COPPER_VIOLATIONS = new Set(['57', '58', '59', '63', '65']);
+
+const COMPLIANCE_STATUS_MAP = new Map([
+  // Indicates the violation is not open but the system has not yet been
+  // designated as returned to compliance.
+  ['K', 'Known'],
+  ['O', 'Open'],
+  ['R', 'Return to compliance'],
+]);
+
+/**
+ * Inserts all rows into the violations table.
+ */
+async function insertRows(db: ConnectionPool, rows: ViolationsTableRow[]): Promise<any[]> {
+  return db.query(sql`INSERT INTO epa_violations (violation_id,
+                                                 pws_id,
+                                                 violation_code,
+                                                 compliance_status,
+                                                 start_date, geom)
+                      VALUES ${sql.join(
+                        rows.map((row: ViolationsTableRow) => {
+                          return sql`(${row.violation_id}, 
+                                      ${row.pws_id}, 
+                                      ${row.violation_code}, 
+                                      ${row.compliance_status}, 
+                                      ${row.start_date},
+                                      ${sql.__dangerous__rawValue(
+                                        `ST_AsText(ST_GeomFromGeoJSON('${row.geom}'))`,
+                                      )})`;
+                        }),
+                        ',',
+                      )};`);
+}
+
+/**
+ * Inserts all rows into the violations table.
+ */
+async function deleteRows(db: ConnectionPool): Promise<any[]> {
+  return db.query(sql`DELETE
+                      FROM epa_violations
+                      WHERE violation_id IS NOT NULL`);
+}
+
+/**
+ * Reads the S3 CSV file and the number of rows successfully written.
+ */
+function parseS3IntoViolationsTableRow(
```

Comment:
will you update to use Brett's common handler in this PR?

Reply:
I think we should wait until I get it working for parcels first.